javascript-bank/transaction.js

```diff
@@ -0,0 +1,14 @@
+/* exported Transaction */
+
+function Transaction(type, amount){
+  this.type = type;
+  this.amount = amount;
+}
+
+Transaction.prototype.type = function () {
+  return this.type;
+}
+
+Transaction.prototype.amount = function () {
+  return this.amount;
+}
```

Comment:
Only the constructor versions of the `type` and `amount` properties of a `Transaction` instance are usable, because JavaScript will stop looking when it sees those. The `prototype` versions are effectively overwritten (or "shadowed", as we call it) by the constructor versions.

Please remove the shadowed versions of the properties.